django/db/models/fields/related.py

```diff
@@ -1885,6 +1885,13 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
+            warnings.warn(
+                "Implicit default of ForeignKey.on_delete has been deprecated. "
```

Comment:
`"on_delete will be a required kwarg for %s in Django 2.0." % self.__class__.__name__`

I assume this should apply to `ForeignObject` as well.